mobile/node.go

```diff
@@ -40,36 +53,45 @@ import (
 	namepb "github.com/ipfs/go-ipfs/namesys/pb"
 	ipath "github.com/ipfs/go-ipfs/path"
 	ipfsconfig "github.com/ipfs/go-ipfs/repo/config"
-	fsrepo "github.com/ipfs/go-ipfs/repo/fsrepo"
-	"github.com/natefinch/lumberjack"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
 	"github.com/op/go-logging"
-	p2phost "gx/ipfs/QmNmJZL7FQySMtE2BQuLMuZg2EB2CLEunJJUSVSc9YnnbV/go-libp2p-host"
-	dht "gx/ipfs/QmRaVcGchmC1stHHK7YhcgEuTk5k1JiGS568pfYWMgT91H/go-libp2p-kad-dht"
-	dhtutil "gx/ipfs/QmRaVcGchmC1stHHK7YhcgEuTk5k1JiGS568pfYWMgT91H/go-libp2p-kad-dht/util"
-	routing "gx/ipfs/QmTiWLZ6Fo5j4KcTVutZJ5KWRRJrbxzmxA4td8NfEdrPh7/go-libp2p-routing"
-	"gx/ipfs/QmTmqJGRQfuH8eKWD1FjThwPRipt1QhqJQNZ8MpzmfAAxo/go-ipfs-ds-help"
-	recpb "gx/ipfs/QmUpttFinNDmNPgFwKN8sZK6BUtBmA68Y4KdSBDXa8t9sJ/go-libp2p-record/pb"
-	proto "gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/proto"
-	peer "gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
-	"io"
-	"io/ioutil"
-	"net"
-	"net/http"
-	"net/url"
-	"path"
-	"time"
 )
 
+// Node configuration structure
 type Node struct {
 	OpenBazaarNode *core.OpenBazaarNode
 	config         NodeConfig
 	cancel         context.CancelFunc
 	ipfsConfig     *ipfscore.BuildCfg
-	apiConfig      *schema.APIConfig
+	apiConfig      *apiSchema.APIConfig
 }
 
-func NewNode(config NodeConfig) (*Node, error) {
+// Mobile empty structure used to carry config data to route-around gomobile's type restrictions
+type Mobile struct{}
```

Comment:
What type restrictions? You mean for gomobile bind? How does this help with them?

Reply:
This is what I posted to @cpacia in the first round of PR comments:

If we try and run gomobile without the changes in this PR, we get:

```
gomobile bind -target=ios/arm64 github.com/OpenBazaar/openbazaar-go/mobile
gomobile: unsupported named type: struct{RepoPath string; AuthenticationToken string; Testnet bool; UserAgent string; WalletTrustedPeer string;DisableWallet bool; DisableExchangerates bool} / *types.Struct
```

When we looked up this error, we found a few places talk about this issue:

https://pliutau.com/gomobile-bind-types/
https://godoc.org/golang.org/x/mobile/cmd/gobind#hdr-Type_restrictions

@hoffmabc did a lot of the work to get this initially up and running, based of Nick's help, so he may be able to provide additional insight